### PR TITLE
[YUNIKORN-1534] Add binpacking test suites

### PR DESCRIPTION
**Pull request by manirajv06**
### What is this PR for?
Added bin packing node sorting policy test suites

### What type of PR is it?
* [ ] - Improvement

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-1534

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

**Comment by manirajv06** [comment]
Found some issues in calculating node available resources and fixed the same. Based on the recent runs, don't see any instability in bin packing test results.

@pbacsko Can you please take a look?

File: test/e2e/bin_packing/bin_packing_test.go [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+		sort.Slice(sortedWorkerNodes1, func(i, j int) bool {
+			node1, node2 := sortedWorkerNodes1[i].Name, sortedWorkerNodes1[j].Name
+			q1, q2 := nodesAvailRes1[node1][domRes], nodesAvailRes1[node2][domRes]
+			return q1.Cmp(q2) == -1
+		})
+		var nodeNames1 []string
+		for _, node := range sortedWorkerNodes1 {
+			nodeNames1 = append(nodeNames1, node.Name)
+		}
+		By(fmt.Sprintf("Sorted nodes by available memory after padding: %v", nodeNames1))
```
Comment (pbacsko) [inline test/e2e/bin_packing/bin_packing_test.go:122]: Nit: no line feed at the end

File: go.mod [hunk 2]
```diff
@@ -41,6 +41,7 @@ require (
 	k8s.io/component-base v0.23.14
 	k8s.io/klog v1.0.0
 	k8s.io/kube-scheduler v0.23.14
+	k8s.io/kubectl v0.0.0 // indirect
```
Comment (pbacsko) [inline go.mod:44]: What's wrong with version like 0.23.14?

File: test/e2e/bin_packing/bin_packing_test.go [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+	"github.com/apache/yunikorn-k8shim/test/e2e/framework/helpers/common"
+	"github.com/apache/yunikorn-k8shim/test/e2e/framework/helpers/k8s"
+)
+
+var _ = Describe("", func() {
+	var ns string
+
+	BeforeEach(func() {
+		ns = "ns-" + common.RandSeq(10)
+		By(fmt.Sprintf("Create namespace: %s", ns))
```
Comment (pbacsko) [inline test/e2e/bin_packing/bin_packing_test.go:40]: Nit: line feed

**Comment by pbacsko** [comment]
LGTM, I just have a question about the version.
Also, please check if the missing "\n" in the `Sprintf()` calls is a problem or not.

**Comment by manirajv06** [comment]
> LGTM, I just have a question about the version. Also, please check if the missing "\n" in the `Sprintf()` calls is a problem or not.

Modified the version. Newline is not required as ginkgo By method takes care of it automatically.

**Comment by pbacsko** [comment]
+1 merging soon

**Pull request closed without merging**